addons/web/static/tests/views/form_tests.js

```diff
@@ -5815,6 +5815,53 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test('form view is not broken if save failed in readonly mode on field changed', function(assert) {
+        assert.expect(5);
+
+        var failFlag = false;
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch:'<form string="Partners">' +
+                    '<header><field name="trululu" widget="statusbar" clickable="True"/></header>' +
+                '</form>',
+            res_id: 1,
+            mockRPC: function (route, args) {
+                if (args.method === 'write' && failFlag) {
+                    assert.step('fail_write');
+                    return $.Deferred().reject();
+                } else if (args.method === 'write') {
+                    assert.step(args.method);
+                }
+                return this._super.apply(this, arguments);
+            }
+        });
+
+        var $selectedState = form.$('.o_statusbar_status button[data-value="4"]');
+        assert.ok($selectedState.hasClass('btn-primary') && $selectedState.hasClass('disabled'),
```

Comment:
use assert.hasClass (no need for messages here)

Reply:
@aab-odoo It's 11.0 branch, assert.hasClass helper is not available